pallets/amm/src/lib.rs

```diff
@@ -337,11 +336,13 @@ pub mod pallet {
                         .base_amount
                         .checked_sub(base_amount)
                         .ok_or(ArithmeticError::Underflow)?;
+
                     liquidity_amount.quote_amount = liquidity_amount
                         .quote_amount
                         .checked_sub(quote_amount)
                         .ok_or(ArithmeticError::Underflow)?;
 
+                    *pool_liquidity_amount = Some(liquidity_amount);
```

Comment:
this is not necessary

Reply:
@GopherJ I believe, this is necessary otherwise this https://github.com/parallel-finance/parallel/blob/amm-refactoring/pallets/amm/src/tests.rs#L269-L271 would fail.

Reply:
@ashi20192020ayush adding `.as_mut` to https://github.com/parallel-finance/parallel/pull/1189/files#diff-081eb769bf1774964cec3ca16e72ce0510198b2c8e604de2b6b9d1bee336e8ccR317 is better. we can remove line 345